aspnet/mvc/controllers/testing.rst

```diff
@@ -1,8 +1,160 @@
-.. include:: /../common/stub-topic.txt
+Testing Controller Logic
+========================
+By `Steve Smith`_
 
-|stub-icon| Testing Controller Logic
-====================================
+Controllers in ASP.NET MVC apps should be small and focused on user-interface concerns. Large controllers that deal with non-UI concerns are more difficult to test and maintain.
 
-.. include:: /../common/stub-notice.txt
+.. contents:: Sections
+	:local:
+	:depth: 1
+	
+`View sample files <https://github.com/aspnet/Docs/tree/master/aspnet/mvc/controllers/testing/sample>`_
 
-.. _issue: https://github.com/aspnet/Docs/issues/123
+Why Test Controllers
+--------------------
+*Controllers* define groups of related *actions*. The grouping of related actions into controllers is useful for :doc:`routing </fundamentals/routing>`, applying :doc:`filters <filters>`, and :doc:`injecting common dependencies <dependency-injection>`. :doc:`Learn more about controllers and actions <actions>`.
```

Comment:
I think you can remove this first paragraph as it is more about what controllers are, which isn't really the focus of this article.